src/client/pages/OfferNew/Checkout/UserDetailsForm.tsx

```diff
@@ -41,9 +41,14 @@ export const UserDetailsForm: React.FC<Props> = ({
 
   // FIXME should we pick which ssn validation to use in a different way? or not really?
   const market = useMarket()
+  const ssnMaxLength = ssnLengthByMarket[market]
   const isValidSsn = createSsnValidator(market!)
 
-  const setSsn = (newSsn: string) => {
+  const handleSsnChange = (event: React.FormEvent<HTMLInputElement>) => {
+    const { value } = event.currentTarget
+    const nonNumbersRegex = /\D/
+    const newSsn = value.replace(nonNumbersRegex, '')
```

Comment:
This is a clean solution and I'm obviously approving it ✨
I would just like to raise a question here: wouldn't we rather check for what we _do_ want in the input rather than for what we _don't_ want in a case like this? To me that would probably be a tiny bit more clear 🙃  You know, something like this:
```suggestion
    const numbersRegex = /^[0-9]*$/
    const newSsn = numbersRegex.test(value) ? value : ''
```

Reply:
I see what you mean, I had a solution like that at first :) but I actually prefer not having the ternary operator. Maybe @palmenhq has influenced me more than I know 😄